tests/ui/implicit_saturating_sub.rs

```diff
@@ -0,0 +1,206 @@
+// run-rustfix
+#![allow(unused_assignments, unused_mut)]
+#![warn(clippy::implicit_saturating_sub)]
+
+fn main() {
+    // Tests for unsigned integers
+
+    let end_8: u8 = 10;
+    let start_8: u8 = 5;
+    let mut u_8: u8 = end_8 - start_8;
+
+    // Lint
+    if u_8 > 0 {
+        u_8 -= 1;
```

Comment:
```suggestion
        u_8 = u_8 - 1;
```

This should be tested at least once (and I think once should be enough).